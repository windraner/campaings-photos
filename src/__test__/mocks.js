export const markersList = [
  {
    id: 'all',
    name: 'Show all',
    isChecked: false,
    type: 'all',
    count: 0
  },
  {
    id: 'pending',
    name: 'Pending',
    isChecked: false,
    type: 'main',
    count: 0
  },
  {
    id: 'approved',
    name: 'Approved',
    isChecked: false,
    type: 'main',
    count: 0
  },
  {
    id: 'rejected',
    name: 'Rejected',
    isChecked: false,
    type: 'main',
    count: 0
  },
  {
    id: 'rightsRequested',
    name: 'Rights requested',
    isChecked: false,
    type: 'main',
    count: 0
  },
  {
    id: 'sticly',
    name: 'Sticly',
    isChecked: false,
    type: 'main',
    count: 0
  },
  {
    id: 'topPhotos',
    name: 'Top photos',
    isChecked: false,
    type: 'secondary',
    count: 0
  },
  {
    id: 'unhealthy',
    name: 'Unhealthy products',
    isChecked: false,
    type: 'secondary',
    count: 0
  },
];
