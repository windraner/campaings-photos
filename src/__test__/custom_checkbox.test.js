import React from 'react';
import CustomCheckbox from '../components/common/custom_checkbox/custom_checkbox';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { markersList } from './mocks';

configure({ adapter: new Adapter() });

const mock = {
  isChecked: false,
  markersList: markersList,
  id: 'pending',
  stateHandler() {},
};

describe('CustomCheckbox', () => {
  it('Check for callback is called', () => {
    const onClickFunc = jest.fn();
    const wrapper = shallow(<CustomCheckbox {...mock} stateHandler={onClickFunc} />);
    const button = wrapper.find('.campaigns-photos__custom-checkbox');
    button.simulate('change');
    expect(onClickFunc).toBeCalled();
  });
  it('matches the snapshot', () => {
    const tree = shallow(<CustomCheckbox {...mock} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it('matches the snapshot', () => {
    const tree = shallow(<CustomCheckbox {...mock} isChecked={true} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
