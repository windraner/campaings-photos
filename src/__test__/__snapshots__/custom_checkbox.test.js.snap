// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCheckbox matches the snapshot 1`] = `
<input
  checked={false}
  className="campaigns-photos__custom-checkbox mr-2"
  onChange={[Function]}
  type="checkbox"
/>
`;

exports[`CustomCheckbox matches the snapshot 2`] = `
<input
  checked={true}
  className="campaigns-photos__custom-checkbox mr-2"
  onChange={[Function]}
  type="checkbox"
/>
`;
